File: backend/finance_app/app/schemas/category.py
from typing import Optional

from pydantic import BaseModel, ConfigDict


class CategoryCreate(BaseModel):
    name: str
    show_analytics: bool = True


class CategoryRead(BaseModel):
    id: int
    name: str
    user_id: int
    show_analytics: bool


class CategoryChange(BaseModel):
    name: Optional[str] = None
    show_analytics: Optional[bool] = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/finances/category.ts
export interface Category {
    id: number;
    name: string;
    user_id: string;
    show_analytics: boolean;
}

export interface CategoryCreate {
    name: string;
    show_analytics?: boolean;
}

export interface CategoryChange {
    name?: string;
    show_analytics?: boolean;
}

export interface CategoryFilter {
    name?: string;
    page?: number;
    size?: number;
}